src/pages/settings/Profile/CustomStatus/StatusPage.js

```diff
@@ -0,0 +1,120 @@
+import React, {useMemo, useCallback, useEffect} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsPropTypes} from '../../../../components/withCurrentUserPersonalDetails';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import StaticHeaderPageLayout from '../../../../components/StaticHeaderPageLayout';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import withLocalize from '../../../../components/withLocalize';
+import MenuItem from '../../../../components/MenuItem';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import * as User from '../../../../libs/actions/User';
+import MobileBackgroundImage from '../../../../../assets/images/money-stack.svg';
+import themeColors from '../../../../styles/themes/default';
+import useLocalize from '../../../../hooks/useLocalize';
+import styles from '../../../../styles/styles';
+import compose from '../../../../libs/compose';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import ROUTES from '../../../../ROUTES';
+
+const propTypes = {
+    ...withCurrentUserPersonalDetailsPropTypes,
+};
+
+function StatusPage({draftStatus, currentUserPersonalDetails}) {
+    const localize = useLocalize();
+    const currentUserEmojiCode = lodashGet(currentUserPersonalDetails, 'status.emojiCode', '');
+    const currentUserStatusText = lodashGet(currentUserPersonalDetails, 'status.text', '');
+    const draftEmojiCode = lodashGet(draftStatus, 'emojiCode');
+    const draftText = lodashGet(draftStatus, 'text');
+
+    const defaultEmoji = draftEmojiCode || currentUserEmojiCode;
+    const defaultText = draftText || currentUserStatusText;
+    const customStatus = defaultEmoji ? `${defaultEmoji} ${defaultText}` : '';
+    const hasDraftStatus = !!draftEmojiCode || !!draftText;
+
+    const updateStatus = useCallback(() => {
+        const endOfDay = new Date();
+        endOfDay.setHours(23, 59, 59, 999);
+        User.updateCustomStatus({text: defaultText, emojiCode: defaultEmoji, clearAfter: endOfDay.toISOString()});
+
+        User.clearDraftCustomStatus();
+        Navigation.goBack(ROUTES.SETTINGS);
+    }, [defaultText, defaultEmoji]);
+
+    const clearStatus = () => {
+        User.clearCustomStatus();
+        User.clearDraftCustomStatus();
+    };
+
+    const footerComponent = useMemo(
+        () =>
+            hasDraftStatus ? (
+                <Button
+                    success
+                    text={localize.translate('statusPage.save')}
+                    onPress={updateStatus}
+                />
+            ) : null,
+        [hasDraftStatus, localize, updateStatus],
+    );
+
+    useEffect(() => () => User.clearDraftCustomStatus(), []);
```

Comment:
do we need here also User.clearCustomStatus();?

Reply:
No need, as it's should remove only draft status when we close the StatusPage without saving the status